src/common/types.ts

```diff
@@ -16,8 +16,16 @@ import type { ZodRawShape, ZodType, ZodTypeAny } from "zod";
 export interface ToolParams {
   title: string;
   summary: string;
-  parameters?: Parameters; // either parameters or a zodSchema should be present
-  zodSchema?: ZodTypeAny;
+  parameters?: Parameters; // either 'parameters' or an 'inputSchema' should be present
+  inputSchema?: ZodTypeAny;
+  /**
+   * Specifies the type of object returned by the tool. <br>
+   * When `outputSchema` is specified, make sure the tool returns `structuredContent` in its callback. <br>
+   * To keep backwards compatibility, the tool's callback can still return a text `content`.
+   *
+   * https://modelcontextprotocol.io/specification/2025-06-18/server/tools#output-schema
+   */
+  outputSchema?: ZodTypeAny;
```

Comment:
How about adding to where we call the tool callback in `server.ts` and adding some logic along the lines of:

* If `structuredContent` is set and `content` isn't, then call `JSON.stringify`
* Throwing an error `outputSchema` is set but `structuredContent` isn't

(You should make sure that `isError` isn't `true` for that.)

Reply:
Makes sense to me, thanks for the suggestion. I just added both here: a97baf2cb0f6a47e0b025efeb1dab54d20b59347